terraform/create.go

```diff
@@ -91,78 +94,56 @@ func (t *Terraform) Create() error {
-						if err != nil && err != io.EOF { // Somehow it gives an io.EOF error every time. TODO: need to debug.
-							mlog.Error("error closing ssh session", mlog.Err(err))
-						}
-					}()
-
-					buf, err := json.Marshal(v)
-					if err != nil {
-						mlog.Error("invalid config", mlog.String("key", k), mlog.Err(err))
-						return
-					}
-					cmd := fmt.Sprintf(`jq '%s = %s' /opt/mattermost/config/config.json > /tmp/mmcfg.json && mv /tmp/mmcfg.json /opt/mattermost/config/config.json`, k, string(buf))
-					if err := session.Run(cmd); err != nil {
-						mlog.Error("error running ssh command", mlog.String("cmd", cmd), mlog.Err(err))
-						return
-					}
-				}()
-			}
+			t.updateConfig(ip, sshc, output)
 
 			// Starting mattermost.
-			session, err := sshc.NewSession()
-			if err != nil {
-				mlog.Error("failed to create session", mlog.Err(err))
+			cmd := fmt.Sprintf(`/opt/mattermost/bin/mattermost &`) // TODO: servicify this.
+			if err := sshc.RunCommand(cmd); err != nil {
+				mlog.Error("error running ssh command", mlog.String("cmd", cmd), mlog.Err(err))
 				return
 			}
-			func() {
-				defer func() {
-					err := session.Close()
-					if err != nil && err != io.EOF { // Somehow it gives an io.EOF error every time. TODO: need to debug.
-						mlog.Error("error closing ssh session", mlog.Err(err))
-					}
-				}()
-
-				cmd := fmt.Sprintf(`/opt/mattermost/bin/mattermost &`) // TODO: servicify this.
-				if err := session.Run(cmd); err != nil {
-					mlog.Error("error running ssh command", mlog.String("cmd", cmd), mlog.Err(err))
-					return
-				}
-			}()
 		}()
 	}
 
 	return nil
 }
 
+func (t *Terraform) updateConfig(ip string, sshc *ssh.Client, output *terraformOutput) {
+	mlog.Info("Updating config", mlog.String("host", ip))
+
+	var dsn string
+	switch t.config.DeploymentConfiguration.DBInstanceEngine {
+	case "postgres":
+		dsn = "postgres://" + t.config.DeploymentConfiguration.DBUserName + ":" + t.config.DeploymentConfiguration.DBPassword + "@" + output.DBEndpoint.Value + "/" + t.config.DeploymentConfiguration.ClusterName + "db?sslmode=disable"
+	case "mysql":
+		dsn = t.config.DeploymentConfiguration.DBUserName + ":" + t.config.DeploymentConfiguration.DBPassword + "@tcp(" + output.DBEndpoint.Value + ")/" + t.config.DeploymentConfiguration.ClusterName + "db?charset=utf8mb4,utf8\u0026readTimeout=30s\u0026writeTimeout=30s"
+	}
+	mlog.Info("dsn: " + dsn) // TODO: remove this later.
+
+	for k, v := range map[string]interface{}{
+		".ServiceSettings.ListenAddress":       ":8065",
+		".ServiceSettings.LicenseFileLocation": "/home/ubuntu/mattermost.mattermost-license",
+		".ServiceSettings.SiteURL":             "http://" + ip + ":8065",
+		".SqlSettings.DriverName":              t.config.DeploymentConfiguration.DBInstanceEngine,
+		".SqlSettings.DataSource":              dsn,
+		".MetricsSettings.Enable":              true,
+		".PluginSettings.Enable":               true,
+		".PluginSettings.EnableUploads":        true,
+	} {
```

Comment:
I like the injection of db and required information but I am wondering if maybe we want to have a base config file for the server that we store in this repository so that the user running the load-test can have more control in customizing an instance without modifying any code. 
That way we could parse everything by leveraging existing code, modify as needed, serialize and upload to target app server.

Reply:
Yeah, I get you. That would be 2 separate ways to edit the config. As you can see, for some keys, we just set values to true. It might be confusing to some users to see different behavior than what they set in their config.json. Then we need to clarify exactly what values are we overriding and document that more prominently. I think the current approach is good enough because it requires minimal user interaction.

Let's visit this when we have a real requirement for this.